services/alerts/src/useAlert.ts

```diff
@@ -1,27 +1,51 @@
-import { useContext } from 'react'
+import { useContext, useRef, useCallback, useMemo } from 'react'
 import { AlertsManagerContext } from './AlertsManagerContext'
-import { Alert, AlertOptions } from './types'
+import {
+    AlertBarComponent,
+    AlertOptions,
+    AlertsManager,
+    AlertsManagerAlert,
+} from './types'
 
 export const useAlert = (
     message: string | ((props: any) => string),
     options: AlertOptions | ((props: any) => AlertOptions) = {}
 ) => {
-    const alertsManager = useContext(AlertsManagerContext)
+    const alertsManager: AlertsManager = useContext(AlertsManagerContext)
+    const ref = useRef<AlertBarComponent>()
```

Comment:
I think this could still easily support multiple alerts per hook (assuming we want that...) if we made it an array and moved the `alert` initialization to the `show` method

Reply:
Resolving this, because we discussed that one alert per hook does make sense and we could add support to showing multiple alerts in a different way once the use-case presents itself.